src/lightning/fabric/strategies/fsdp.py

```diff
@@ -267,6 +270,28 @@ def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         torch.distributed.broadcast_object_list(obj, src, group=_group.WORLD)
         return obj[0]
 
+    def clip_gradients_norm(
+        self,
+        optimizer,
+        max_norm: Union[float, int],
+        norm_type: Union[float, int] = 2.0,
+        error_if_nonfinite: bool = True,
+    ):
+        """Clip gradients by norm."""
+        from lightning.fabric.wrappers import _unwrap_objects
+
+        optimizer = _unwrap_objects(optimizer)
```

Comment:
@justusschock We will need a `Fabric.clip_gradients()` method too. Its responsibility will be to pass the real optimizer to the strategy instead of the _FabricOptimizer. The strategy is designed to always operate on the non-fabric wrapped objects.

Reply:
I will move the unwrapping out of here then :)